### PR TITLE
[wasm] Build Wasm.Build.Tests only on the WBT job, on CI

### PR DESCRIPTION
**Pull request by radical**
On the `_WasmBuildTests` job, we build the library tests also, which aren't used at all. So, disable that. And we don't need to build Wasm.Build.Tests for the regular build job, so disable there.

This saves 10-12 mins of build time.

### CODE REVIEW

**Review by lewing — COMMENTED** [review]
It couldn't you set TestAssemblies=false and avoid the wasm specific condition in the libraries setup?

**Comment by radical** [comment]
> It couldn't you set TestAssemblies=false and avoid the wasm specific condition in the libraries setup?

It is used in other cases, like the linker tests (see https://github.com/dotnet/runtime/blob/main/eng/pipelines/libraries/execute-trimming-tests-steps.yml#L7, and https://github.com/dotnet/runtime/blob/main/eng/pipelines/runtime-linker-tests.yml#L79). The intention is clearer with having an explicit property.

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -324,7 +324,7 @@ jobs:
     jobParameters:
       testGroup: innerloop
       nameSuffix: AllSubsets_Mono_WasmBuildTests
-      buildArgs: -s mono+libs+host+packs+libs.tests -c $(_BuildConfig) /p:ArchiveTests=true
+      buildArgs: -s mono+libs+host+packs+libs.tests -c $(_BuildConfig) /p:ArchiveTests=true /p:OnlyWasmBuildTests=true
```
Comment (lewing) [inline eng/pipelines/runtime.yml:327]:
```suggestion
      buildArgs: -s mono+libs+host+packs+libs.tests -c $(_BuildConfig) /p:ArchiveTests=true /p:OnlyWasmBuildTests=true /p:TestAssemblies=false
```

I meant this here then remove the new condition from src/libraries/tests.proj

**Comment by radical** [comment]
`runtime (Libraries Test Run release coreclr windows x64 Debug)` is https://github.com/dotnet/runtime/issues/56810

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
I'd kind of prefer TestWasmBuild=true and dropping the only idea but it isn't a strong preference

**Comment by radical** [comment]
> I'd kind of prefer TestWasmBuild=true and dropping the only idea but it isn't a strong preference

With the `TestAssemblies=false` change, your suggestion makes sense. I will add it to another PR that I have open.

**Pull request merged**